hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/repository/upsert/TokenAccountUpsertQueryGenerator.java

```diff
@@ -39,47 +38,60 @@
     private static final String JOIN_TABLE = "token";
     private final String finalTableName = "token_account";
     private final String temporaryTableName = getFinalTableName() + "_temp";
-    private final List<String> v1ConflictIdColumns = List.of(TokenAccountId_.TOKEN_ID, TokenAccountId_.ACCOUNT_ID);
-    // createdTimestamp is needed for v2 schema compliance as it's used in index
-    private final List<String> v2ConflictIdColumns = List.of(TokenAccountId_.TOKEN_ID,
-            TokenAccountId_.ACCOUNT_ID, TokenAccount_.CREATED_TIMESTAMP);
-    private final Set<String> nonUpdatableColumns = Set.of(TokenAccountId_.ACCOUNT_ID,
-            TokenAccount_.CREATED_TIMESTAMP, TokenAccount_.ID, TokenAccountId_.TOKEN_ID);
+    private final Set<String> nonUpdatableColumns = Set.of(TokenAccountId_.ACCOUNT_ID, TokenAccount_.ASSOCIATED,
+            TokenAccount_.AUTO_ASSOCIATED, TokenAccount_.CREATED_TIMESTAMP, TokenAccount_.ID, TokenAccountId_.TOKEN_ID);
 
     @Getter(lazy = true)
+    @SuppressWarnings("java:S3740")
     // JPAMetaModelEntityProcessor does not expand embeddedId fields, as such they need to be explicitly referenced
-    private final Set<SingularAttribute> selectableColumns = Set.of(TokenAccountId_.accountId,
-            TokenAccount_.associated, TokenAccount_.createdTimestamp, TokenAccount_.freezeStatus,
+    private final Set<SingularAttribute> selectableColumns = Set.of(TokenAccountId_.accountId, TokenAccount_.associated,
+            TokenAccount_.autoAssociated, TokenAccount_.createdTimestamp, TokenAccount_.freezeStatus,
             TokenAccount_.kycStatus, TokenAccount_.modifiedTimestamp, TokenAccountId_.tokenId);
 
     @Override
-    public String getInsertWhereClause() {
+    protected String getDeleteWhereClause() {
+        // a token dissociate and a subsequent token associate for the same <token, account> can happen in the same
+        // batch; per the logic in SqlEntityListener, the associate will override the token dissociate while the
+        // previous <token, account> association is still in db; so the condition here should be
+        //   where token_account_temp.associated is not null
```

Comment:
Thank you for the detailed comment.  Minor nit:
```suggestion
        // where token_account_temp.associated is not null
```

Reply:
it's done intentionally to emphasis the where clause

Reply:
Got it.  I guess an alternative could be capitalizing the `WHERE` in this case, but I'm fine with this.